pkg/console/subresource/configmap/configmap.go

```diff
@@ -66,25 +66,34 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 // - a bool indicating if config was merged (unsupportedConfigOverrides)
 // - an error
 func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, managedConfig *corev1.ConfigMap, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
+	// There are multiple sources for the config
+	//1. default brand `okd`, as well as default doc url & logout url set
+	//2. extract managed config and apply on top of this, without nilling out prev values
+	//3. apply operator config on top of this, without nilling out prev values
+	//4. apply unsupported config overrides on top of this, but should not nil out prev values
+
+	// Build a default config that can be overwritten from other sources
+	host := rt.Spec.Host
+	apiServerURL := getApiUrl(infrastructureConfig)
+	defaultConfig := NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL)
+
+	// Derive a config using operator and console configs
```

Comment:
Do the helpers matter anymore? or can we just inline:

```
brand := operatorConfig.Spec.Customization.Brand  // its either a value or ""
```

Reply:
(because `len(str)` is just checking for an empty string anyway)

Reply:
sounds good